src/interactivity/base/RemoteConsoleControl.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "RemoteConsoleControl.hpp"
+
+#include "../../inc/HostSignals.hpp"
+
+using namespace Microsoft::Console::Interactivity;
+
+RemoteConsoleControl::RemoteConsoleControl(HANDLE signalPipe) :
+    _pipe{ signalPipe }
+{
+}
+
+#pragma region IConsoleControl Members
+
+template<typename T>
+[[nodiscard]] NTSTATUS _SendTypedPacket(HANDLE pipe, ::Microsoft::Console::HostSignals signalCode, T& payload)
+{
+    // To ensure it's a happy wire format, pack it tight at 1.
+#pragma pack(push, 1)
+    struct HostSignalPacket
+    {
+        ::Microsoft::Console::HostSignals code;
+        T data;
+    };
+#pragma pack(pop)
+
+    HostSignalPacket packet;
+    packet.code = signalCode;
+    packet.data = payload;
+
+    DWORD dwWritten = 0;
+    if (!WriteFile(pipe, &packet, sizeof(packet), &dwWritten, nullptr))
+    {
+        return NTSTATUS_FROM_WIN32(::GetLastError());
+    }
+
+    return STATUS_SUCCESS;
+}
+
+[[nodiscard]] NTSTATUS RemoteConsoleControl::NotifyConsoleApplication(_In_ DWORD dwProcessId)
+{
+    HostSignalNotifyAppData data = { 0 };

```

Comment:
In C this was necessary, but in C++ you can just write `data{}`.
The implicitly declared default constructor for any class or struct performs zero initialization since C++11.

Reply:
Huh. OK.